ts/src/pro/poloniexfutures.ts

```diff
@@ -0,0 +1,741 @@
+'use strict';
+
+//  ---------------------------------------------------------------------------
+
+import poloniexfuturesRest from '../poloniexfutures.js';
+import { AuthenticationError } from '../base/errors.js';
+import { ArrayCache, ArrayCacheBySymbolById } from '../base/ws/Cache.js';
+import { Precise } from '../base/Precise.js';
+
+//  ---------------------------------------------------------------------------
+
+export default class poloniexfutures extends poloniexfuturesRest {
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'has': {
+                'ws': true,
+                'watchOHLCV': false,
+                'watchOrderBook': true,
+                'watchTicker': true,
+                'watchTickers': false,
+                'watchTrades': true,
+                'watchBalance': true,
+                'watchStatus': false,
+                'watchOrders': true,
+                'watchMyTrades': false,
+            },
+            'urls': {
+                'api': {
+                    'ws': 'wss://futures-apiws.poloniex.com/endpoint',
+                },
+            },
+            'options': {
+                'tradesLimit': 1000,
+                'ordersLimit': 1000,
+                'watchOrderBookMethod': '/contractMarket/level3v2', // can also be '/contractMarket/level2'
+                'publicToken': undefined,
+                'privateToken': undefined,
+            },
+        });
+    }
+
+    async getPublicToken (params = {}) {
+        if (this.options['publicToken'] === undefined) {
+            const response = await this.publicPostBulletPublic ();
+            //
+            //    {
+            //        code: '200000',
+            //        data: {
+            //            instanceServers: [ [Object] ],
+            //            token: 'DcXijCbKcWFew_i0BS8y6UNmBtlHW3UAvR4Nx4VADIn15tt-jDqMbYWNZ2II5fSnrClCBBv6dTDc8PMFHz-H6tSnN_vkspYYmOImrn5NXLlsFbcpggjU6mMGfZAja_q_-wgHjBcT7RhvJVwiLf8PgR2VF0_UPbEwl-RWj6JCmic=.9NqWNqQVILOkGiD1RL1AoQ=='
+            //        }
+            //    }
+            //
+            const data = this.safeValue (response, 'data');
+            this.options['publicToken'] = this.safeString (data, 'token');
+        }
+        return this.options['publicToken'];
+        // TODO: deal with expired token?
```

Comment:
I'm not sure how I should be handling expired tokens.

Reply:
It doesn't look like the expiration time is provided, I think that comment is here because I initially copied the poloniex rest api, and the rest docs might mention it